packages/slate-editor/src/extensions/placeholders/elements/GalleryBookmarkPlaceholderElement.tsx

```diff
@@ -72,61 +66,45 @@ export function GalleryBookmarkPlaceholderElement({
 
     usePlaceholderManagement(element.type, element.uuid, {
         onTrigger: handleTrigger,
+        // @ts-expect-error Figure out how to fix this
         onResolve: handleData,
     });
 
+    useEffect(() => {
+        if (!isSelected) {
+            setCustomRendered(false);
+        }
+    }, [isSelected]);
+
     return (
-        <SearchInputPlaceholderElement<NewsroomGallery>
-            {...props}
+        <PlaceholderElement
+            attributes={attributes}
             element={element}
-            // Core
             format={format}
             icon={PlaceholderGallery}
             title="Click to insert a media gallery bookmark"
             description="Add a link to your media gallery"
-            // Input
-            getSuggestions={getSuggestions}
-            renderAddon={renderAddon}
-            renderEmpty={renderEmpty}
-            renderSuggestion={renderSuggestion}
-            renderSuggestions={(props) => (
-                <SearchInput.Suggestions
-                    activeElement={props.activeElement}
-                    query={props.query}
-                    suggestions={props.suggestions}
-                    footer={renderSuggestionsFooter?.(props)}
-                    origin={props.origin}
-                >
-                    {props.children}
-                </SearchInput.Suggestions>
-            )}
-            inputTitle="Media gallery bookmark"
-            inputDescription="Add a media gallery card to your stories, campaigns and pitches"
-            inputPlaceholder="Search for media galleries"
-            onSelect={handleSelect}
+            onClick={() => setCustomRendered(true)}
+            overflow="visible"
+            renderFrame={
+                isCustomRendered
+                    ? () => renderPlaceholder({ onRemove: handleRemove, onSelect: handleSelect })
+                    : undefined
+            }
```

Comment:
I am still wondering if it's a good idea to have this half-approach where the "initial" placeholder UI is rendered by the editor, but then the subsequent UI is custom rendered.

I think ideally it would be nice if we could render the whole placeholder outside of the editor (i.e. rendered via `renderPlaceholder` function) but this would also mean that we would need to implement drag & drop functionality and other things (loading states, placeholder management etc.), so I wasn't sure if we should do it now or at all.

My thinking was that the placeholder extension would be a "headless" extension, only providing the necessary logic (callbacks etc.), but then expecting the consumer to implement the UI for it.

What do you think?